foundry/applications/basePlaceableHUD.d.ts

```diff
@@ -1,29 +1,91 @@
-// @TODO:
-
 /**
  * An abstract base class for displaying a heads-up-display interface bound to a Placeable Object on the canvas
  */
-declare class BasePlaceableHUD extends Application {
+declare abstract class BasePlaceableHUD<
+  P extends PlaceableObject<PlaceableObject.Data>,
+  L extends PlaceablesLayer<P>
+> extends Application {
+  constructor(...args: ConstructorParameters<typeof Application>);
+
   /**
-   * BasePlaceableHUD display states
+   * Reference a PlaceableObject this HUD is currently bound to
    */
-  DISPLAY_STATES: object;
+  object: P | null;
 
   /**
-   * Reference a PlaceableObject this HUD is currently bound to
+   * @override
+   * @defaultValue
+   * ```
+   * mergeObject(super.defaultOptions, {
+   *   classes: ["placeable-hud"],
+   *   popOut: false
+   * })
+   * ```
    */
-  object: PlaceableObject;
+  static get defaultOptions(): Application.Options;
 
-  constructor(...args: any[]);
+  /**
+   * Convenience access for the canvas layer which this HUD modifies
+   */
+  get layer(): L;
 
   /**
    * Bind the HUD to a new PlaceableObject and display it
    * @param object - A PlaceableObject instance to which the HUD should be bound
    */
-  bind(object: PlaceableObject): void;
+  bind(object: P): void;
 
   /**
    * Clear the HUD by fading out it's active HTML and recording the new display state
    */
   clear(): void;
+
+  /** @override */
+  _render(...args: Parameters<Application['_render']>): Promise<void>;
+
+  /**
+   * @override
+   * @defaultValue
+   * ```
+   * mergeObject(this.object.data, {
+   *   id: this.id,
+   *   classes: this.options.classes.join(' '),
+   *   appId: this.appId,
+   *   isGM: game.user.isGM,
+   *   icons: CONFIG.controlIcons
+   * });
+   * ```
+   */
+  getData(
+    options?: Application.RenderOptions
+  ): P['data'] & {
+    id: Application['id'];
+    classes: string;
+    appId: Application['appId'];
+    isGM: boolean;
+    icons: typeof CONFIG['controlIcons'];
+  };
+
+  /** @override */
+  setPosition({ left, top, width, height, scale }?: Partial<Application.Position>): any;
```

Comment:
does this return `any`  because it doesn't return anything but `Application.setPosition` does? If so, please add a remark an consider filing a bug with foundryvtt.